raft/src/test/java/org/apache/kafka/raft/KafkaRaftClientTest.java

```diff
@@ -1059,14 +1151,14 @@ public void testFetchShouldBeTreatedAsLeaderEndorsement() throws Exception {
 
         int findQuorumCorrelationId = assertSentFindQuorumRequest();
         channel.mockReceive(new RaftResponse.Inbound(findQuorumCorrelationId,
-            findQuorumResponse(localId, epoch, voters), -1));
+            findQuorumResponse(OptionalInt.of(localId), epoch, voters), -1));
 
         pollUntilSend(client);
 
         // We send BeginEpoch, but it gets lost and the destination finds the leader through the FindQuorum API
         assertBeginQuorumEpochRequest(epoch);
 
-        FetchQuorumRecordsRequestData fetchRequest = fetchQuorumRecordsRequest(epoch, otherNodeId, 0L);
+        FetchQuorumRecordsRequestData fetchRequest = fetchQuorumRecordsRequest(epoch, otherNodeId, 0L, 0);
```

Comment:
We don't have to address this in the current PR, but I feel a lot of request/response data generation code paths are overlapping with the core KafkaRaftClient library. Potentially we could consolidate them by putting the data initialization logic into their corresponding request/response classes, like moving this call towards `FetchQuorumRecordsRequest.java`

Reply:
Yeah, I have the same feeling. I was trying hard not to use objects like `FetchQuorumRecordsRequest` in this implementation because ultimately we will need to remove them. However, frankly speaking, the use of builders all over the place makes for very verbose code. I was thinking instead of moving these helpers into `RaftUtil`.